### PR TITLE
Various invariants fixes

### PR DESCRIPTION
**Pull request by grandizzy**
- fix `F1` invariant failing due to rounding in local fenwick-like tree. Always update deposit at index before a pool action that removes from deposit (remove quote token, move quote token, kick with deposit). `test_regression_removeQuote_F1` regression test
- fix `B5` invariant failing due to wrong settle invariant logic (`test_regression_settle_buckets_B5` regression test)
    - update borrower deposit time when settle and no deposit in pool (hence borrower compensated with LP)
    - do not update borrower deposit time when settle bad debt (hence no LP awarded to borrower)
    - update borrower deposit time in compensated bucket when take settle NFT auction and fraction of NFT compensated with LP in auction bucket
-  Make min / max amount of debt to take in single action configurable (`MIN_DEBT_AMOUNT`, `MAX_DEBT_AMOUNT`). Default min debt amount 1e3, max debt amount to 1e28

### CODE REVIEW

**Review by EdNoepel — COMMENTED** [review]
I'm fine with `MAX_DEBT_AMOUNT`, but not `MIN_DEBT_AMOUNT`.  We need to ensure pool mechanics function if users draw an abnormally small amount of debt.

**Comment by grandizzy** [comment]
> I'm fine with `MAX_DEBT_AMOUNT`, but not `MIN_DEBT_AMOUNT`. We need to ensure pool mechanics function if users draw an abnormally small amount of debt.

Can you confirm if I properly understand: you mean to remove `MIN_DEBT_AMOUNT` config and leave it hardcoded to 1e3, is it his correct?

**Comment by EdNoepel** [comment]
> > I'm fine with `MAX_DEBT_AMOUNT`, but not `MIN_DEBT_AMOUNT`. We need to ensure pool mechanics function if users draw an abnormally small amount of debt.
> 
> Can you confirm if I properly understand: you mean to remove `MIN_DEBT_AMOUNT` config and leave it hardcoded to 1e3, is it his correct?

No; I don't believe invariant tests should restrict debt to any minimum amount other than the minimum debt amount enforced by the protocol.

**Comment by grandizzy** [comment]
> > > I'm fine with `MAX_DEBT_AMOUNT`, but not `MIN_DEBT_AMOUNT`. We need to ensure pool mechanics function if users draw an abnormally small amount of debt.
> > 
> > 
> > Can you confirm if I properly understand: you mean to remove `MIN_DEBT_AMOUNT` config and leave it hardcoded to 1e3, is it his correct?
> 
> No; I don't believe invariant tests should restrict debt to any minimum amount other than the minimum debt amount enforced by the protocol.

The issue trying to solve is to run for larger period of time / higher depth. If we keep max debt for example to 1e30 (which is unrealistic imo for a single draw debt action) we will be able to run with something like 1500 bucket depth max. Also other kind of addition could reduce this number as it can be seen in when fixing PROTOTECH-2:https://github.com/ajna-finance/contracts/actions/runs/4912168210/jobs/8770898372

**Review by prat-gpt — APPROVED** [review]
LGTM.

**Comment by grandizzy** [comment]
> > > > I'm fine with `MAX_DEBT_AMOUNT`, but not `MIN_DEBT_AMOUNT`. We need to ensure pool mechanics function if users draw an abnormally small amount of debt.
> > > 
> > > 
> > > Can you confirm if I properly understand: you mean to remove `MIN_DEBT_AMOUNT` config and leave it hardcoded to 1e3, is it his correct?
> > 
> > 
> > No; I don't believe invariant tests should restrict debt to any minimum amount other than the minimum debt amount enforced by the protocol.
> 
> The issue trying to solve is to run for larger period of time / higher depth. If we keep max debt for example to 1e30 (which is unrealistic imo for a single draw debt action) we will be able to run with something like 1500 bucket depth max. Also other kind of addition could reduce this number as it can be seen in when fixing PROTOTECH-2:https://github.com/ajna-finance/contracts/actions/runs/4912168210/jobs/8770898372

I got this wrong, I just noticed you said you're OK with `MAX_DEBT_AMOUNT` but not with `MIN_DEBT_AMOUNT`, will remove such

**Comment by grandizzy** [comment]
> > > > > I'm fine with `MAX_DEBT_AMOUNT`, but not `MIN_DEBT_AMOUNT`. We need to ensure pool mechanics function if users draw an abnormally small amount of debt.
> > > > 
> > > > 
> > > > Can you confirm if I properly understand: you mean to remove `MIN_DEBT_AMOUNT` config and leave it hardcoded to 1e3, is it his correct?
> > > 
> > > 
> > > No; I don't believe invariant tests should restrict debt to any minimum amount other than the minimum debt amount enforced by the protocol.
> > 
> > 
> > The issue trying to solve is to run for larger period of time / higher depth. If we keep max debt for example to 1e30 (which is unrealistic imo for a single draw debt action) we will be able to run with something like 1500 bucket depth max. Also other kind of addition could reduce this number as it can be seen in when fixing PROTOTECH-2:https://github.com/ajna-finance/contracts/actions/runs/4912168210/jobs/8770898372
> 
> I got this wrong, I just noticed you said you're OK with `MAX_DEBT_AMOUNT` but not with `MIN_DEBT_AMOUNT`, will remove such

kept `MIN_DEBT_AMOUNT` configurable but set default to 0 in https://github.com/ajna-finance/contracts/pull/796/commits/91e0e506223699c161b1e6ca6986c8fac745b56b

File: tests/forge/regression/ERC721Pool/RegressionTestReservesERC721Pool.t.sol [hunk 1]
```diff
@@ -265,7 +265,7 @@ contract RegressionTestReserveERC721Pool is ReserveERC721PoolInvariants {
     }
 
     /*
-        Test was failing due to the fact invariant paralele logic was calculating borrow fee rate using post action interest rate.
+        Test was failing due to the fact invariant paralel logic was calculating borrow fee rate using post action interest rate.
```
Comment (EdNoepel) [inline tests/forge/regression/ERC721Pool/RegressionTestReservesERC721Pool.t.sol:268]: getting closer; `parallel`

**Review by EdNoepel — APPROVED** [review]
Just a spelling issue.

**Pull request merged**